server/src/main/java/org/elasticsearch/common/time/JavaDateFormatter.java

```diff
@@ -61,27 +63,34 @@
         if (distinctLocales > 1) {
             throw new IllegalArgumentException("formatters must have the same locale");
         }
+        this.printer = printer;
+        this.format = format;
         if (parsers.length == 0) {
             this.parsers = new DateTimeFormatter[]{printer};
         } else {
             this.parsers = parsers;
         }
-        this.format = format;
-        this.printer = printer;
     }
 
     @Override
-    public TemporalAccessor parse(String input) {
-        DateTimeParseException failure = null;
+    public TemporalAccessor parse(final String input) {
+        ElasticsearchParseException failure = null;
         for (int i = 0; i < parsers.length; i++) {
             try {
                 return parsers[i].parse(input);
             } catch (DateTimeParseException e) {
+                String msg = "could not parse input [" + input + "] with date formatter [" + format + "]";
```

Comment:
If possible I would extract that into createExceptionMessage method or similar. This would make the catch block a bit more concise

Reply:
done